File: py/search/array/JumpSearch.py
#!/usr/bin/env python
# -*- coding:gbk -*-

import math
from typing import List

class JumpSearch:
    def search(self, nums: List[int], target: int) -> int:
        step = int(math.sqrt(len(nums)))
        border = step
        while border < len(nums) and target > nums[border]:
            border += step

        for i, value in enumerate(nums[border - step : min(border + 1, len(nums))]):
            if target == value:
                return border - step + i
        return -1

File: java/search/array/JumpSearch.java
package search.array;

public class JumpSearch {
    public int search(int[] nums, int target) {
        int step = (int) Math.sqrt(nums.length);
        int border = step;
        while (border < nums.length && target > nums[border]) {
            border += step;
        }

        for (int i = border - step; i < Math.min(border + 1, nums.length); ++i) {
            if (target == nums[i]) {
                return i;
            }
        }
        return -1;
    }
}
